sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/aggregator/DoubleExponentialHistogramBuckets.java

```diff
@@ -306,7 +308,20 @@ public boolean equals(@Nullable Object obj) {
    * </ul>
    */
   private boolean sameBucketCounts(DoubleExponentialHistogramBuckets other) {
+    if (this.totalCount != other.totalCount) {
+      return false;
+    }
     int min = Math.min(this.counts.getIndexStart(), other.counts.getIndexStart());
+
+    // This check is so we avoid iterating from Integer.MIN_VALUE.
+    // In this case, we can assume that those buckets are empty.
+    // We start iterating from the other bucket index instead.
+    // They still may be equal as it is possible for another set of buckets
+    // to be empty but have a higher start index.
+    if (min == Integer.MIN_VALUE) {
+      min = Math.max(this.counts.getIndexStart(), other.counts.getIndexStart());
+    }
```

Comment:
This problem kind of stems from `AdaptingCircularBufferCounter.isEmpty()` not returning correct result after a `diff()` that makes it empty. However I'm not sure if there's a way to fix it without putting in an _O(n)_ check somewhere in that class.  

This works well as a workaround and brings the `diffAccumulation()` test down from 16s to 0.002s.